functional_data/lib/src/annotations.dart

```diff
@@ -1,7 +1,37 @@
 import 'package:collection/collection.dart';
 
 class FunctionalData {
-  const FunctionalData();
+  const FunctionalData({
+    this.generateCopy,
+    this.generateCopyWith,
+    this.generateCopyUsing,
+    this.generateLenses,
+  });
+
+  /// Whether the generator should generate the copy methods.
+  ///
+  /// When set, it overrides the settings of [generateCopyWith] and [generateCopyUsing].
```

Comment:
Instead of overriding, I'd prefer it to be incompatible and have asserts on the constructor.

Reply:
Same thing for `generateLenses: true` and `generateCopyWith: false`?